classes/Cart.php

```diff
@@ -4281,6 +4281,14 @@ public function duplicate()
         $success = true;
         $products = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS('SELECT * FROM `' . _DB_PREFIX_ . 'cart_product` WHERE `id_cart` = ' . (int) $this->id);
 
+        foreach ($products as $key => $product) {
+            $currentProduct = new Product();
+            $currentProduct->hydrate($product);
+            if ($currentProduct->hasAttributes() && $product['id_product_attribute'] === 0) {
+                unset($products[$key]);
```

Comment:
Hi @202-ecommerce 
I'm worried about this solution, because:
- it will result in potentially empty carts, I wonder if this can cause other issues later
- (considering the initial issue) it doesn't fix the problem when you have a Cart containing a former simple product, the you cal still finish your order

I think we'd rather need a validation of the cart during the checkout process, or even in the cart preview, which prevents from ordering this kind of product.

Reply:
@jolelievre we think it's not a problem to continue without the unavailable product because we consider it's not good to stop reorder with one product unavailable.

In this case @tlhuillier propose to add a error notification. We suggest a red alert like #11631 ?
"You cannot reorder the product xxx because it is not available any more."

So is this solution better for you ?

Reply:
Hi @202-ecommerce,
indeed I think it's better to notify the user that something is wrong. The error message you mention in the linked PR seems a good solution. So I see two possibilities:
- remove the unavailable product and display an error notification to notify the user that one product was removed
- leave the unavailable product but display a warning notifying that the cart can't be checkout because the user needs to select a combination or he removes this product

The second solution seems more user friendly to me because:
- the user can still access the product and select the appropriate combination (if you remove it he needs to take time finding the right product again..)
- what happens if you remove all the products from the cart (including when there's only one), you end up with an empty cart which can be confusing

Reply:
In case we agree on leaving the unavailable products but with an error, of course the checkout button should be blocked, and the checkout tunnel inaccessible (redirect to the cart)
I realize this solution is longer to develop but I think it's a better experience for the user